lib/dfe/analytics/load_entity_batch.rb

```diff
@@ -5,33 +5,44 @@ class LoadEntityBatch < AnalyticsJob
       # at a batch size of 500, this allows 20kb per record
       BQ_BATCH_MAX_BYTES = 10_000_000
 
-      def perform(model_class_arg, ids)
-        # Support string args for Rails < 6.1
-        model_class = if model_class_arg.respond_to?(:constantize)
-                        model_class_arg.constantize
-                      else
-                        model_class_arg
-                      end
-
-        events = model_class.where(id: ids).map do |record|
-          DfE::Analytics::Event.new
-            .with_type('import_entity')
-            .with_entity_table_name(model_class.table_name)
-            .with_data(DfE::Analytics.extract_model_attributes(record))
-        end
+      def perform(model_class_arg, ids, import_entity_id)
+        model_class = resolve_model_class(model_class_arg)
+
+        events = create_events(model_class, ids, import_entity_id)
 
         payload_byte_size = events.sum(&:byte_size_in_transit)
 
         # if we overrun the max batch size, recurse with each half of the list
         if payload_byte_size > BQ_BATCH_MAX_BYTES
           ids.each_slice((ids.size / 2.0).round).to_a.each do |half_batch|
             Rails.logger.info "Halving batch of size #{payload_byte_size} for #{model_class.name}"
-            self.class.perform_later(model_class_arg, half_batch)
+            self.class.perform_later(model_class_arg, half_batch, import_entity_id)
           end
         else
-          DfE::Analytics::SendEvents.perform_now(events.as_json)
+          DfE::Analytics::SendEvents.perform_now(events)
+        end
+      end
+
+      private
+
+      def resolve_model_class(model_class_arg)
+        # Support string args for Rails < 6.1
+        model_class_arg.respond_to?(:constantize) ? model_class_arg.constantize : model_class_arg
+      end
+
+      def create_events(model_class, ids, import_entity_id)
+        model_class.where(id: ids).map do |record|
+          build_event(record, model_class.table_name, import_entity_id)
         end
       end
+
+      def build_event(record, table_name, import_entity_id)
+        DfE::Analytics::Event.new
+            .with_type('import_entity')
+            .with_entity_table_name(table_name)
+            .with_tags(import_entity_id)
```

Comment:
@ericaporter will this be sufficient to allow us to tell the difference between import events from two different imports even if they are for the same entity?

Reply:
Looking at line 16 of lib/dfe/analytics/tasks/import_entities.rake I think it would be? But if so maybe we could call it something different? I thought this was referring to the UID of an entity, or the name of an entity (e.g. application_choices) rather than the UID of the import. Maybe entity_import_id or something?

Reply:
Happy to rename if it makes things easier

Reply:
I suppose the name `import_entity_id` may be confusing, but it doesn't actually get added to the event. Only the value gets added as a tag. The value is a numeric date BTW.

What about `import_entities_job_id` OR just `import_job_id` ?